homeassistant/components/camera/img_util.py

```diff
@@ -78,10 +78,10 @@ class TurboJPEGSingleton:
     seconds.
     """
 
-    __instance = None
+    __instance: TurboJPEG | Literal[False] | None = None
 
     @staticmethod
-    def instance() -> TurboJPEG:
+    def instance() -> TurboJPEG | Literal[False] | None:
```

Comment:
Returning `None` doesn't look possible.

Reply:
True, however `mypy` doesn't know that. It can't infer that calling `TurboJEPGSingleton()` will set `__instance`, thus it uses the type from the declaration above which is initialized with `None`.

I could add an assert but that seems redundant as any caller needs to handle `False` already.
```py
assert TurboJPEGSingleton.__instance is not None
```
https://github.com/home-assistant/core/blob/63971385892b12d4eef2124999e19c450a2aec0b/homeassistant/components/camera/img_util.py#L45-L47

https://github.com/home-assistant/core/blob/63971385892b12d4eef2124999e19c450a2aec0b/homeassistant/components/stream/core.py#L486-L487

Reply:
Ok.